projects/storefrontlib/shared/services/import-export/import-csv.service.ts

```diff
@@ -0,0 +1,91 @@
+import { Injectable } from '@angular/core';
+import { ValidationErrors } from '@angular/forms';
+import { Observable } from 'rxjs';
+import { map, tap } from 'rxjs/operators';
+import { InvalidFileInfo } from '../../models/file';
+import { ImportService } from './import.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ImportCsvService {
+  constructor(protected importService: ImportService) {}
+  /**
+   * Processes the CSV data
+   *
+   * @param csvString raw extracted data from CSV
+   * @param ignoreHeader flag allows for ignore headers row while reading
+   * @param separator for csv data
+   * @returns Processed data containing productCode and quantity
+   */
+  readCsvData(
+    csvString: string,
+    separator: string,
+    ignoreHeader = true
+  ): string[][] {
+    return csvString
+      .split('\n')
+      .map((row) => row.split(separator).map((cell) => cell.replace(/"/g, '')))
+      .filter(
+        (value, index) => !(ignoreHeader && index === 0) && value[0] !== ''
+      );
+  }
+
+  loadCsvData(file: File, separator: string): Observable<string[][]> {
+    return this.importService
+      .loadTextFile(file)
+      .pipe(map((res) => this.readCsvData(res as string, separator)));
+  }
+
+  validateData(
```

Comment:
We don't validate data, but the whole file. What about renaming to `validateFile()`?